src/Features/LanguageServer/Protocol/Handler/DocumentChanges/DidChangeHandler.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.DocumentChanges
+{
+    [Shared]
+    [ExportLspMethod(LSP.Methods.TextDocumentDidChangeName, mutatesSolutionState: true)]
+    internal class DidChangeHandler : IRequestHandler<LSP.DidChangeTextDocumentParams, object>
+    {
+        private readonly ILspSolutionProvider _solutionProvider;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public DidChangeHandler(ILspSolutionProvider solutionProvider)
+        {
+            _solutionProvider = solutionProvider;
+        }
+
+        public LSP.TextDocumentIdentifier? GetTextDocumentIdentifier(LSP.DidChangeTextDocumentParams request) => request.TextDocument;
+
+        public async Task<object> HandleRequestAsync(LSP.DidChangeTextDocumentParams request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var documents = _solutionProvider.GetDocuments(request.TextDocument.Uri, context.ClientName);
```

Comment:
I'm starting to see why the solution provider is a bit annoying

I wonder if instead of this here, we could just use the context's document, then call http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/Document.cs,6155e64747e478e6 to retrieve the linked ones and iterate that way.

It might make the code a bit clearer on what it's doing (even if it's the same thing)

Reply:
Thank you for pointing me to that, I'm always happy to remove another use of solution provider :)

Reply:
No more solution provider in any DidXXX handlers :D